app/models/event.rb

```diff
@@ -3,9 +3,12 @@ class Event < ActiveRecord::Base
   belongs_to :student, foreign_key: :user_id
   belongs_to :activity
   belongs_to :creator, class_name: "Teacher"
-  has_many :scans, dependent: :destroy
 
   def scanned_in?
-    !scans.where(correct: true).empty?
+    scans.where('timestamp > ? AND timestamp < ?', start_time, end_time).present?
+  end
+
+  def scans
+    Scan.where(correct: true, student:student, location: location)
```

Comment:
as for scans, correct or not, that is done when the [scan is created](https://github.com/rootselementary/check_in_tool_rails/blob/2cb38520f45dd8cac0a6d3abd582f9f674176e4a/app/models/scan_creator.rb#L7). Good catch on fixing up the schema. I'll add that to the migration.